src/gwas/src/gwas/plots/worker.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+from multiprocessing import Pool
+from pathlib import Path
+from typing import List
+
+import blosc2
+import pandas as pd
+from tqdm import tqdm
+
+from gwas.plots.helpers import chi2_pvalue, find_phenotype_index, load_metadata
+
+
+def create_dataframe_single_chr(args: List[tuple[Path, Path, str]]) -> pd.DataFrame:
+    """
+    Generates a pandas DataFrame for genetic variants of a single chromosome.
+
+    Args:
+        args: A tuple containing three elements:
+            - score_path (Path): The path to the chromosome's score file.
+            - axis_metadata_path (Path): The path to the chromosome's axis metadata file.
+            - phenotype_label (str): The label of the phenotype to calculate p-values for.
+
+    Returns:
+        pd.DataFrame: A DataFrame with columns ['SNP', 'CHR', 'BP', 'P'], containing the
+        SNP identifier, chromosome number, base pair position, and p-value for each variant.
+
+    Raises:
+        ValueError: If the length of the b2 array does not match the length of the metadata.
+    """
+    # score_path: str | Path, axis_metadata_path: str | Path, phenotype_label: str old args
+    score_path, axis_metadata_path, phenotype_label = args
+
+    axis_metadata = load_metadata(axis_metadata_path)
+    metadata = pd.DataFrame(axis_metadata[0])
+    phenotypes_list = axis_metadata[1]
+
+    b2_array = blosc2.open(urlpath=score_path)
+
+    if b2_array.shape[0] != len(metadata.index):
+        raise ValueError(
+            "The length of b2_array does not match the length of the metadata."
+        )
+
+    u_stat_idx, v_stat_idx = find_phenotype_index(
+        phenotype_label=phenotype_label, phenotypes_list=phenotypes_list
+    )
+
+    variant_count = len(metadata.index)
+
+    u_stat, v_stat = (
+        b2_array[:, u_stat_idx],
+        b2_array[:, v_stat_idx],
+    )  # possible solution b2_array[:, u_stat_idx:v_stat_idx] to only index array once
```

Comment:
Please don't forget to check this :-)

Reply:
`    stats = b2_array[
        :, u_stat_idx : v_stat_idx + 1
    ]`